File: web/lib/utils/organization.ts
export function getOrganizationAbbreviation(name: string): string {
  if (!name) return 'O'
  
  const words = name.trim().split(/\s+/).filter(word => word.length > 0)
  
  if (words.length === 0) return 'O'
  
  if (words.length === 1) {
    const word = words[0]
    if (word.length <= 3) {
      return word.toUpperCase()
    }
    return word.substring(0, 3).toUpperCase()
  }
  
  return words
    .slice(0, 3)
    .map(word => word.charAt(0).toUpperCase())
    .join('')
}
export interface StandardizedOrganizationData {
  name: string
  organization_type: string
  admin_email: string
  phone_number: string | null
  description: string | null
  status?: string
  logo_url?: string | null
}

export function standardizeOrganizationData(data: {
  name: string
  organization_type: string
  admin_email: string
  phone_number?: string | null
  description?: string | null
  status?: string
  logo_url?: string | null
}): StandardizedOrganizationData {
  const trimmedName = (data.name || '').trim()
  const trimmedOrgType = (data.organization_type || '').trim()
  const trimmedAdminEmail = (data.admin_email || '').trim()
  
  const phoneNumber = data.phone_number 
    ? (data.phone_number.trim() || null)
    : null
  
  const description = data.description 
    ? (data.description.trim() || null)
    : null
  
  const status = (data.status && data.status.trim()) || 'pending'
  
  return {
    name: trimmedName,
    organization_type: trimmedOrgType,
    admin_email: trimmedAdminEmail,
    phone_number: phoneNumber,
    description: description,
    status: status,
    logo_url: data.logo_url || null,
  }
}

File: mobile/lib/core/utils/organization.dart
/// Organization utility functions matching web version

class OrganizationUtils {
  /// Get organization abbreviation from name
  /// Matches web: getOrganizationAbbreviation
  static String getOrganizationAbbreviation(String name) {
    if (name.isEmpty) return 'O';
    
    final words = name.trim().split(RegExp(r'\s+')).where((word) => word.isNotEmpty).toList();
    
    if (words.isEmpty) return 'O';
    
    if (words.length == 1) {
      final word = words[0];
      if (word.length <= 3) {
        return word.toUpperCase();
      }
      return word.substring(0, 3).toUpperCase();
    }
    
    return words
        .take(3)
        .map((word) => word.isNotEmpty ? word[0].toUpperCase() : '')
        .join('');
  }

  /// Standardize organization data
  /// Matches web: standardizeOrganizationData
  static Map<String, dynamic> standardizeOrganizationData({
    required String name,
    required String organizationType,
    required String adminEmail,
    String? phoneNumber,
    String? description,
    String? status,
    String? logoUrl,
  }) {
    final trimmedName = name.trim();
    final trimmedOrgType = organizationType.trim();
    final trimmedAdminEmail = adminEmail.trim();
    
    final phone = phoneNumber?.trim();
    final desc = description?.trim();
    final orgStatus = (status?.trim().isNotEmpty ?? false) ? status!.trim() : 'pending';
    
    return {
      'name': trimmedName,
      'organization_type': trimmedOrgType,
      'admin_email': trimmedAdminEmail,
      'phone_number': phone?.isNotEmpty == true ? phone : null,
      'description': desc?.isNotEmpty == true ? desc : null,
      'status': orgStatus,
      'logo_url': logoUrl,
    };
  }
}
